inference-engine/src/cldnn_engine/cldnn_program.cpp

```diff
@@ -2919,6 +2922,71 @@ void Program::CreateSoftMaxPrimitive(cldnn::topology& topology, InferenceEngine:
     AddPrimitiveToProfiler(softmaxLayerName, layer);
 }
 
+void Program::CreateLogSoftmaxPrimitive(cldnn::topology& topology, InferenceEngine::CNNLayerPtr &layer) {
+    ValidateLayer(layer, 1);
+    auto inputPrimitives = GetPrevLayersPrimitives(layer);
+    auto logSoftmaxLayer = as<InferenceEngine::GenericLayer*>(layer);
+    auto sz = logSoftmaxLayer->input().get()->getTensorDesc().getDims().size();
+
+    auto axis = logSoftmaxLayer->GetParamAsInt("axis", 1);
+    if (axis < 0) axis += sz;
+    std::vector<uint16_t> reduce_axes(1);
+
+    switch (axis) {
+        case 0: {
+            reduce_axes = { cldnn::reduce::reduce_axis::along_b,
+                            cldnn::reduce::reduce_axis::along_f,
+                            cldnn::reduce::reduce_axis::along_y,
+                            cldnn::reduce::reduce_axis::along_x };
+            if (sz > 4) reduce_axes.push_back(cldnn::reduce::reduce_axis::along_z);
```

Comment:
Why do you do reduction over all axes for axis=0? I suppose it should be along_b